yoga/Yoga.cpp

```diff
@@ -4059,12 +4098,15 @@ void YGNodeCalculateLayout(
     const float ownerWidth,
     const float ownerHeight,
     const YGDirection ownerDirection) {
+  uint16_t depth = 0;
```

Comment:
It would be better to use `int` for the depth type, so as not to limit to a tree height of 65,535.  (The generation count, a separate thing, is likely OK to wrap at 16-bits.)

To simplify this further you might consider removing the variable declaration, and writing it directly as an argument when invoking `YGLayoutNodeInternal` below.  For example:
```
  if (YGLayoutNodeInternal(
          node,
          width,
          height,
          ownerDirection,
          widthMeasureMode,
          heightMeasureMode,
          ownerWidth,
          ownerHeight,
	  true,
	  "initial",
	  node->getConfig(),
	  0, // tree root
	  generationCount)) {
```